### PR TITLE
[quick_actions] Android support only calling initialize once

### PR DESCRIPTION
**Pull request by monmanuela**
Fixes flutter/flutter#87259

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

File: packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java
Comment (stuartmorgan) [inline packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java:21]: Why is this static? Wouldn't each instance of the plugin need to be initialized?

File: packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java
Comment (stuartmorgan) [inline packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java:51]: Why is this (and the new method) needed? I don't see any deferment logic, so couldn't it just send the call unconditionally?

File: packages/quick_actions/quick_actions_platform_interface/lib/method_channel/method_channel_quick_actions.dart
Comment (stuartmorgan) [inline packages/quick_actions/quick_actions_platform_interface/lib/method_channel/method_channel_quick_actions.dart:28]: If you're going to call a new method, you should make it a no-op on iOS to avoid getting useless error messages (although per other comments, it's not clear to me that this is necessary).

**Review by stuartmorgan — COMMENTED** [review]
> * [x]  I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.

While this does change a test file, there's no actually testing of the new functionality. Per separate discussion, that will need a native integration test that actually drives a quick actions flow.

**Comment by stuartmorgan** [comment]
Adding @blasten to review the Android intent flow aspect of this.

File: packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java
Comment (blasten) [inline packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java:74]:
```suggestion
    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N_MR1) {
      return false;
    }
```

File: packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java
Comment (blasten) [reply packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java:21]: sounds like the use case is to support a single initialization.  If that is the case, it could still be an instance field. 

That said, I don't there's such a case as registering/initializing the same plugin name more than once.

File: packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java
Comment (blasten) [inline packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java:83]: MethodCallHandlerImpl could take an object that implements an interface. Then, QuickActionsPlugin could implement the interface.  Lastly, pass the reference when it constructs `MethodCallHandlerImpl`.

Comment (blasten) [reply packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java:83]: The interface implementation sets `isInitialized = true;`

File: packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java
Comment (blasten) [inline packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java:51]: could we add comments to the code? That will help the next maintainer

Comment (monmanuela) [reply packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java]: Ah my bad. Initially I thought it'll call onNewIntent when the Dart side isn't initialized yet, which means it'll make a redundant method call which would just vanish. But after checking again, the method call does go through to Dart.

Removed the if block, and the new method. Thanks for catching this.

File: packages/quick_actions/quick_actions_platform_interface/lib/method_channel/method_channel_quick_actions.dart
Comment (monmanuela) [reply packages/quick_actions/quick_actions_platform_interface/lib/method_channel/method_channel_quick_actions.dart]: Removed as it's not necessary.

File: packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java
Comment (monmanuela) [reply packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java]: Thanks, done.

File: packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java
Comment (monmanuela) [reply packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java:82]: Thanks for the suggestion. But as per stuartmorgan@'s comment above, removed the field as it's not necessary.

File: packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java
Comment (monmanuela) [reply packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java]: Removed as it's unnecessary. Thanks.

File: packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java
Comment (monmanuela) [reply packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java]: Done.

**Comment by monmanuela** [comment]
Thanks for the reviews! 😄 Responded to the comments.

**Review by blasten — APPROVED** [review]
Implementation LGTM, but it may also need a test.

@bparrishMines is there a plan to backfill this plugin as well?

**Comment by monmanuela** [comment]
Per @bparrishMines 's suggestions in a separate discussion, added unit tests for the new functionality

**Review by stuartmorgan — APPROVED** [review]
It wasn't actually clear to me from the offline discussion why unit tests are better here than UI tests of the actual flow (which would ensure that whole user flow behaves as expected), but it looks like this has reasonable coverage of the change so LGTM.

**Pull request merged**
